#! /bin/sh /usr/share/dpatch/dpatch-run
## 04-apigen.dpatch

@DPATCH@

--- codespeak-lib-0.9.0.orig/py/apigen/htmlgen.py
+++ codespeak-lib-0.9.0/py/apigen/htmlgen.py
@@ -178,6 +178,8 @@
         rev = None
         if wc.check(versioned=True):
             rev = py.path.svnwc(packageroot).info().rev
+        else:
+            rev = py.__package__.revision
         _revcache[packageroot] = rev
     if packageroot.basename == "py": 
         assert rev is not None
--- codespeak-lib-0.9.0.orig/py/apigen/tracer/docstorage.py
+++ codespeak-lib-0.9.0/py/apigen/tracer/docstorage.py
@@ -21,8 +21,17 @@
     for key, value in defs.iteritems():
         chain = key.split('.')
         base = module
-        for elem in chain:
-            base = getattr(base, elem)
+        try:
+            for elem in chain:
+                base = getattr(base, elem)
+        except RuntimeError, exc:
+            if elem == "greenlet":
+                print exc.__class__.__name__, exc
+                print "Greenlets not supported on this platform. Skipping apigen doc for this module"
+                continue
+            else:
+                raise
+
         if value[1] == '*':
             d.update(get_star_import_tree(base, key))
         else:
--- codespeak-lib-0.9.0.orig/py/bin/_docgen.py    2007-02-14 11:13:06.000000000 +0000
+++ codespeak-lib-0.9.0/py/bin/_docgen.py        2008-02-05 11:09:29.000000000 +0000
@@ -28,7 +28,7 @@
 def build_docs(targetpath, testargs):
     docpath = pypath.join('doc')
     run_tests(docpath, '',
-              testargs + ' --forcegen --apigenrelpath="apigen/"')
+              testargs + ' --forcegen --apigen="%s/apigen/apigen.py"' % (pypath,))
     docpath.copy(targetpath)

